x-pack/plugins/alerting/public/alert_api.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { HttpSetup } from 'kibana/public';
+import * as t from 'io-ts';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { fold } from 'fp-ts/lib/Either';
+import { findFirst } from 'fp-ts/lib/Array';
+import { isNone } from 'fp-ts/lib/Option';
+
+import { i18n } from '@kbn/i18n';
+import { BASE_ALERT_API_PATH, alertStateSchema } from '../common';
+import { Alert, AlertType, AlertTaskState } from '../common';
+
+export async function loadAlertTypes({ http }: { http: HttpSetup }): Promise<AlertType[]> {
```

Comment:
Yeah, I hate the duplication but this seems to be the approach in Kibana - we don't reuse code between plugins.

TBH I think this api *should* reside under `plugins/alerting` and triggers_ui should be importing the alerting public api and using it there.

Thoughts?

Reply:
Lets move the discussion about this here: https://github.com/elastic/kibana/issues/59892

Reply:
Going to have to defer to @mikecote and @YulNaumenko on this.  I'm sure I asked at one point why we created triggers_actions_ui instead of putting those bits in alerts and actions, there must have been a reasonable answer for that.  I'm slightly worried that we'll have ui code for alerts now spread across two plugins.  And it kind of makes the discuss issue about re-org a foregone conclusion, at least eliminates the option of "let's keep everything in triggers_actions_ui", if that ends up being the decision.

Reply:
There might be a really great reason why we HAVE to have this in a new alerting/public folder as well, but not seeing it ATM.

Reply:
Mike convinced me https://github.com/elastic/kibana/issues/59892#issuecomment-598722472 that trying out the alerting/public for this is a worthwhile effort :-)